testutil-server/src/main/java/io/spine/server/TestBoundedContexts.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright 2018, TeamDev Ltd. All rights reserved.
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.server;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.protobuf.Message;
+import io.spine.client.TestActorRequestFactory;
+import io.spine.core.Command;
+import io.spine.core.Event;
+import io.spine.core.TenantId;
+import io.spine.grpc.MemoizingObserver;
+import io.spine.server.event.EventStreamQuery;
+import io.spine.server.tenant.TenantAwareOperation;
+import io.spine.util.Exceptions;
+
+import java.util.List;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static io.spine.Identifier.newUuid;
+import static io.spine.grpc.StreamObservers.memoizingObserver;
+
+/**
+ * Utilities for tests that deal with {@link BoundedContext Bounded Context}.
+ * 
+ * @author Mykhailo Drachuk
+ */
+@VisibleForTesting
+public class TestBoundedContexts {
```

Comment:
Right now it's a bit of a mess. Those are just the methods joined together. 

Instead we need a helper for writing the integration tests (or something). Like an API for white-box / black-box testing approaches. We are going to invent it :) Let's discuss in person.